lib/perfdata/elasticsearchdatastreamwriter.hpp

```diff
@@ -0,0 +1,115 @@
+/* Icinga 2 | (c) 2012 Icinga GmbH | GPLv2+ */
+
+#ifndef ELASTICSEARCHDATASTREAMWRITER_H
+#define ELASTICSEARCHDATASTREAMWRITER_H
+
+#include <boost/beast/http/verb.hpp>
+#include <boost/beast/http/status.hpp>
+
+#include "base/configobject.hpp"
+#include "base/dictionary.hpp"
+#include "base/shared-object.hpp"
+#include "base/workqueue.hpp"
+#include "base/timer.hpp"
+#include "base/tlsstream.hpp"
+#include "config/expression.hpp"
+#include "icinga/checkable.hpp"
+#include "icinga/checkresult.hpp"
+#include "icinga/macroprocessor.hpp"
+#include "remote/url.hpp"
+
+#include "perfdata/elasticsearchdatastreamwriter-ti.hpp"
+
+namespace icinga
+{
+
+class EcsDocument final : public SharedObject {
+	public:
+		DECLARE_PTR_TYPEDEFS(EcsDocument);
+
+		EcsDocument() = default;
+		EcsDocument(String index, Dictionary::Ptr document);
+
+		String GetIndex() const { return m_Index; }
+		void SetIndex(const String& index) { m_Index = index; }
+		Dictionary::Ptr GetDocument() const { return m_Document; }
+		void SetDocument(Dictionary::Ptr document) { m_Document = document; }
+
+	private:
+		String m_Index;
+		Dictionary::Ptr m_Document;
+};
+
+class ElasticsearchDatastreamWriter final : public ObjectImpl<ElasticsearchDatastreamWriter>
+{
+public:
+	DECLARE_OBJECT(ElasticsearchDatastreamWriter);
+	DECLARE_OBJECTNAME(ElasticsearchDatastreamWriter);
+
+	static void StatsFunc(const Dictionary::Ptr& status, const Array::Ptr& perfdata);
+
+	static String FormatTimestamp(double ts);
+
+	void ValidateHostTagsTemplate(const Lazy<Array::Ptr> &lvalue, const ValidationUtils &utils) override;
+	void ValidateServiceTagsTemplate(const Lazy<Array::Ptr> &lvalue, const ValidationUtils &utils) override;
+	void ValidateHostLabelsTemplate(const Lazy<Dictionary::Ptr> &lvalue, const ValidationUtils &utils) override;
+	void ValidateServiceLabelsTemplate(const Lazy<Dictionary::Ptr> &lvalue, const ValidationUtils &utils) override;
+	void ValidateFilter(const Lazy<Value> &lvalue, const ValidationUtils &utils) override;
+
+protected:
+	void OnConfigLoaded() override;
+	void Resume() override;
+	void Pause() override;
+	Value TrySend(const Url::Ptr& url, String body);
+
+private:
+	WorkQueue m_WorkQueue{10000000, 1};
+	boost::signals2::connection m_HandleCheckResults;
+	Timer::Ptr m_FlushTimer;
+	bool m_Paused = false;
```

Comment:
You don't need this attribute as there's an existing one in one of the base classes. You can just replace all its usages with `GetPaused()`.